app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt

```diff
@@ -141,14 +141,13 @@ class PreviewBadge : View {
                     }
                 }
 
+                val checkListLength = checkList[i].list.size
+                val checkListHeight = checkList.size
                 when (badgeMode) {
                     Mode.LEFT -> {
                         val animationValue = animationIndex.div(200)
-                        if (validMarquee || flashLEDOn &&
-                                i < checkList.size &&
-                                j < checkList[i].list.size &&
-                                j >= animationValue &&
-                                checkList[i].list[j - animationValue]) {
+                        if (validMarquee || flashLEDOn && i < checkList.size && j < checkList[i].list.size &&
+                                checkList[i].list[(checkListLength + j - animationValue).rem(checkListLength)]) {
```

Comment:
It was clearer before, arrange as it already was

Reply:
I have done that